app/models/user_email.rb

```diff
@@ -22,6 +22,23 @@ class UserEmail < ActiveRecord::Base
   before_save -> { destroy_email_tokens(self.email_was) }, if: :will_save_change_to_email?
 
   after_destroy { destroy_email_tokens(self.email) }
+  def self.ensure_consistency!
+    user_ids_without_primary_email = DB.query_single <<~SQL
+      SELECT u.id
+      FROM users u
+      LEFT JOIN user_emails ue ON u.id = ue.user_id AND ue.primary = true
+      WHERE ue.id IS NULL;
+    SQL
```

Comment:
Just curious how this is possible? When creating a user, do we not create the `UserEmail` and the `User` record in the same DB transaction?

Reply:
Unfortunately, we don't actually know how the data ended up in this state, just that it could and ended up blocking rebake of old posts. Same thing with the user avatars missing users.